.github/workflows/ci.yml

```diff
@@ -7,6 +7,13 @@ on:
   pull_request:
   merge_group:
 
+# Ensures that only one job is allowed to run at any given time for a pull
+# request (or commit sha). Existing jobs if the PR for example is rebased or
+# force pushed are cancelled.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}
+  cancel-in-progress: true
```

Comment:
Ah! This is nice - I was thinking about adding this, but I wanted multiple merges into main to each get their own job. I like the "PR | sha" form to solve this problem